scripts/globals/functional.lua

```diff
@@ -0,0 +1,95 @@
+-----------------------------------
```

Comment:
Everything you have in this file should be put in `utils.lua` and attached to the `xi` object.

Reply:
will do. so xi.utils rather than utils?

Reply:
Done